src/main/java/com/zendesk/maxwell/metrics/MaxwellHealthCheck.java

```diff
@@ -1,29 +1,24 @@
 package com.zendesk.maxwell.metrics;
 
-import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Meter;
 import com.codahale.metrics.health.HealthCheck;
-import com.zendesk.maxwell.producer.AbstractAsyncProducer;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.zendesk.maxwell.producer.AbstractProducer;
 
 
 public class MaxwellHealthCheck extends HealthCheck {
+	private final Meter failedMessageMeter;
 
-	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellHealthCheck.class);
-
-	private MetricRegistry metricRegistry;
-
-	public MaxwellHealthCheck(MetricRegistry metricRegistry) {
-		this.metricRegistry = metricRegistry;
+	public MaxwellHealthCheck(AbstractProducer producer) {
+		this.failedMessageMeter = producer.getFailedMessageMeter();
 	}
 
 	@Override
 	protected Result check() throws Exception {
 		// TODO: this should be configurable.
-		if (this.metricRegistry.getMeters().get(AbstractAsyncProducer.failedMessageMeterName).getFifteenMinuteRate() > 0) {
+		if (failedMessageMeter != null && failedMessageMeter.getFiveMinuteRate() > 0) {
 			return Result.unhealthy(">1 messages failed to be sent to Kafka in the past 15minutes");
```

Comment:
a little late to the game, but i think the text here needs updating ("15minutes" -> "5minutes")